gammapy/irf/energy_dispersion.py

```diff
@@ -541,6 +545,291 @@ def gauss_energy_dispersion_matrix(ebounds, sigma=0.2):
+            raise ValueError("Energies must be Quantity objects.")
+        if not isinstance(offset_lo, Angle) or not isinstance(offset_hi, Angle):
+            raise ValueError("Offsets must be Angle objects.")
+
+        self.migra_lo = migra_lo
+        self.migra_hi = migra_hi
+        self.offset_lo = offset_lo
+        self.offset_hi = offset_hi
+        self.dispersion = dispersion
+
+        self.ebounds = EnergyBounds.from_lower_and_upper_bounds(etrue_lo, etrue_hi)
+        self.energy = self.ebounds.log_centers
+        self.offset = (offset_hi + offset_lo) / 2
+        self.migra = (migra_hi + migra_lo) / 2
+
+        self._prepare_linear_interpolator()
+
+    @classmethod
+    def from_fits(cls, hdu):
+        """Create `EnergyDispersion2D` from ``GCTAEdisp2D`` format HDU.
+
+        Parameters
+        ----------
+        hdu : `~astropy.io.fits.BinTableHDU`
+            ``ENERGY DISPERSION`` extension.
+            
+        """
+
+        data = hdu.data
+        header = hdu.header
+        e_lo = EnergyBounds(data['ETRUE_LO'].squeeze(), header['TUNIT1'])
+        e_hi = EnergyBounds(data['ETRUE_HI'].squeeze(), header['TUNIT2'])
+        o_lo = Angle(data['THETA_LO'].squeeze(), header['TUNIT5'])
+        o_hi = Angle(data['THETA_HI'].squeeze(), header['TUNIT6'])
+        m_lo = data['MIGRA_LO'].squeeze()
+        m_hi = data['MIGRA_HI'].squeeze()
+        matrix = data['MATRIX'].squeeze()
+
+        return cls(e_lo, e_hi, m_lo, m_hi, o_lo, o_hi, matrix)
+
+    @classmethod
+    def read(cls, filename):
+        """Create `EnergyDispersion2D` from ``GCTAEdisp2D`` format FITS file.
+
+        Parameters
+        ----------
+        filename : str
+            File name
+        """
+        hdulist = fits.open(filename)
+        return cls.from_fits(hdulist['ENERGY DISPERSION'])
+
+    def evaluate(self, offset=None, e_true=None, migra=None):
+        """Probability for a given offset, true energy, and migration
+        """
+
+        if offset is None:
+            offset = self.offset
+        if e_true is None:
+            e_true = self.energy
+        if migra is None:
+            migra = self.migra
+
+        if not isinstance(e_true, Quantity):
+            raise ValueError("Energy must be a Quantity object.")
+        if not isinstance(offset, Angle):
+            raise ValueError("Offset must be an Angle object.")
+
+        offset = offset.to('degree')
+        e_true = e_true.to('TeV')
+
+        val = self._eval(offset=offset, e_true=e_true, migra=migra)
+
+        return val
+
+    def _eval(self, offset=None, e_true=None, migra=None):
+
+        x = np.asarray(offset.value)
+        y = np.asarray(migra)
+        z = np.asarray(np.log10(e_true.value))
+        ax = [x, y, z]
+
+        in_shape = (ax[0].size, ax[1].size, ax[2].size)
+
+        for i, s in enumerate(ax):
+            if ax[i].shape == ():
+                ax[i] = ax[i].reshape(1)
+
+        # There is a bug here that could be investigated
```

Comment:
Add "TODO: " here or a new Github issue.
Otherwise we'll never find / remember this.